### PR TITLE
Ask PRs to annotate tests

### PR DESCRIPTION
**Pull request by max-sixty**
- [x] Passes `pre-commit run --all-files`
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

As discussed https://github.com/pydata/xarray/pull/5690#issuecomment-897280353

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!-- Feel free to remove check-list items aren't relevant to your change -->
 
 - [ ] Closes #xxxx
-- [ ] Tests added
+- [ ] Tests added, and functions annotated with `-> None:`
```
Comment (shoyer) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: Let's discuss this more back in #5690, but I don't think I would understand what this means from the description here alone.

**Comment by mathause** [comment]
An alternative would be to use [check_untyped_defs](https://mypy.readthedocs.io/en/stable/config_file.html#confval-check_untyped_defs). However, this would be much more strict. (I tried that once but ran into mypy errors I could not fix, so I gave up.)

**Comment by max-sixty** [comment]
> An alternative would be to use [check_untyped_defs](https://mypy.readthedocs.io/en/stable/config_file.html#confval-check_untyped_defs). However, this would be much more strict. (I tried that once but ran into mypy errors I could not fix, so I gave up.)

Ah nice! That would be better. I guess we want it for new code — it would be the herculean effort to fix the old code, most of which doesn't imply a true problem.

FWIW I think most of the false positives are when reassinging variables to different types; e.g. `expected =`. I guess someone could write a script to increment each new `expected =` to `expected2 =` etc, but quite the mission as well.

Is there a way of explaining the TODO in a way that's easy for new contributors to understand? That's probably the closest solution here

**Comment by max-sixty** [comment]
As discussed on the call — can we get mypy coverage?

**Comment by dcherian** [comment]
Here are the docs: https://mypy.readthedocs.io/en/stable/command_line.html#report-generation

**Comment by max-sixty** [comment]
Nice, thanks @dcherian .

Is anyone familiar with coveralls / whether this can be added? Is it possible to have a rule like "_new_ functions need to be typed", by looking at the diff? That requires storing the previous results.

One other alternative is to do the work to get all the test files typed, and then enforce @mathause 's idea of `check_untyped_defs` on those files only.

**Pull request closed without merging**